c/examples/v4/cpp_example/cpp_example.cc

```diff
@@ -0,0 +1,93 @@
+#include <iostream>
+#include <fstream>
+
+#include <libsbp/cpp/state.h>
+#include <libsbp/cpp/message_handler.h>
+
+void usage(char *prog_name) {
+  fprintf(stderr, "usage: %s [filename]\n", prog_name);
+}
+
+class SbpFileReader : public sbp::IReader {
+ public:
+  SbpFileReader(const char *file_path) : file_stream_(file_path, std::ios::binary | std::ios_base::in) { }
+
+  bool is_open() const { return file_stream_.is_open(); }
+  bool eof() const { return file_stream_.eof(); }
+
+  s32 read(u8 *buffer, u32 buffer_length) override {
+    auto start_index = file_stream_.tellg();
+    if (start_index == -1) {
+      return -1;
+    }
+    file_stream_.read(reinterpret_cast<char *>(buffer), buffer_length);
+    auto end_index = file_stream_.tellg();
+    if (end_index == -1 || file_stream_.fail()) {
+      return -1;
+    }
+
+    return static_cast<s32>(end_index - start_index);
+  }
+
+ private:
+  std::ifstream file_stream_;
+};
+
+class ECEFHandler : private sbp::MessageHandler<sbp_msg_gps_time_t, sbp_msg_pos_ecef_t> {
+  public:
+    ECEFHandler(sbp::State *state) : sbp::MessageHandler<sbp_msg_gps_time_t, sbp_msg_pos_ecef_t>(state) {
+    }
+
+    void handle_sbp_msg(uint16_t sender_id, const sbp_msg_gps_time_t& msg) {
+      (void)sender_id;
+      std::cout << "Received new GPS_TME message with WN = " << msg.wn << ", TOW = " << msg.tow << "\n";
+    }
+
+    void handle_sbp_msg(uint16_t sender_id, const sbp_msg_pos_ecef_t& msg) {
+      (void)sender_id;
+      std::cout << "Received new POS_ECEF message with TOW = " << msg.tow;
+      std::cout << ", (X,Y,Z) = (" << msg.x << "," << msg.y << "," << msg.z << ")\n";
+    }
+};
+
+class EverythingHandler : private sbp::AllMessageHandler {
```

Comment:
The legacy example also includes a `LLHFrameHandler` frame handler but as we don't support that any more (if I've understood correctly?) I just removed it (otherwise implementing it as a `MessageHandler` would make it look fairly identical to the existing `ECEFHandler`).